data_transfer/devices/dreem.py

```diff
@@ -88,53 +96,56 @@ def download_metadata(self) -> None:
-                    device_record = dreem_devices[0]
-                # Edge-case: multiple dreem headbands used, e.g., if one broke.
-                elif len(dreem_devices) > 1:
-                    # Determine usage based on weartime as it exists in UCAM
-                    device_record = ucam.record_by_wear_period_in_list(
-                        dreem_devices, recording.start, recording.end
-                    )
-                # Edge-case: device not logged with patient in UCAM
-                else:
-                    print(f"Metadata cannot be determined for:\n    {recording}")
-                    continue
-            else:
+            if not (patient_id := utils.format_id_patient(_patient_id)):
+                log.error(
+                    f"Record NOT created: Error formatting PatientID ({_patient_id}) for\n{recording}\n"
+                )
                 unknown += 1
-                print(f"Metadata cannot be determined for:\n    {recording}")
                 continue
 
+            device_id = self.__device_id_from_ucam(
+                patient_id, recording.start, recording.end
+            ) or inventory.device_id_by_serial(device_serial)
+
+            if not patient_id or not device_id:
+                log.error(f"Metadata cannot be determined for {recording}")
+                log.error(f"Patient ({patient_id}) or Device ID ({device_id}) missing.")
+                unknown += 1
+                continue  # Skip record
+            known += 1
+
             record = Record(
-                filename=recording.id,
-                device_type=self.device_type,
+                hash=hash_id,
+                manufacturer_ref=recording.id,
+                device_type=self.device_type.name,
                 patient_id=patient_id,
-                **asdict(device_record),
+                device_id=device_id,
+                start_wear=recording.start,
+                end_wear=recording.end,
             )
 
             create_record(record)
-            print(f"Record Created:\n   {record}")
 
-            path = Path(config.storage_vol / f"{record.filename}-meta.json")
             # Store metadata from memory to file
-            utils.write_json(path, item)
+            utils.write_json(record.metadata_path(), item)
+
+        log.debug(f"{known} records created and {unknown} NOT this session.")
 
-            print(f"Metadata saved to: {path}\n")
-        print(f"{known} records created and {unknown} NOT this session.")
+    def __unknown_records(self, records: List[Dict]) -> Dict[str, Dict]:
+        """
+        Only add records that are not known in the DB, i.e., ID and filename.
+        """
+        results = {}
+        known_records = all_hashes()
+        for record in records:
+            record_hash = uid_to_hash(record["id"], self.device_type)
+            if record_hash not in known_records:
+                results[record_hash] = record
+        return results
```

Comment:
I think it is a fair solution to write this method per-device pipeline as they focus on different keys in the `item` object. We can look at pulling this out along with #24 to ensure the result is equal across pipelines.